src/Avalonia.X11/X11Window.cs

```diff
@@ -763,6 +737,38 @@ private void OnPropertyChange(IntPtr atom, bool hasValue)
 
         }
 
+        private WindowState AtomsToWindowState(IntPtr nitems, IntPtr* pitems)
+        {
+            for (var c = 0; c < nitems.ToInt32(); c++)
```

Comment:
Can you please replace this with a single iteration and set local variable flags minimized/fullscreen/maximized? Then check the flags after the loop and return the correct state.